python/cudf/cudf/core/frame.py

```diff
@@ -1434,22 +1442,18 @@ def _get_sorted_inds(
         Get the indices required to sort self according to the columns
         specified in by.
         """
-
-        to_sort = [
-            *(
-                self
-                if by is None
-                else self._get_columns_by_label(list(by), downcast=False)
-            )._columns
-        ]
+        if by is None:
+            to_sort = self._columns
+        else:
+            to_sort = self._get_columns_by_label(list(by))._columns
```

Comment:
Outside of dropping of `downcast`, is there any reason for the restructure here? Are there cases where we would expect the unpacking here not to work?

Reply:
This was mostly a micro-optimization to avoid the unnecessary unpacking of the `if by is None` case